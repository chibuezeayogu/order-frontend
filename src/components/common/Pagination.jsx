import React from 'react'

const Pagination = ({ ordersPerPage, totalOrders, paginate }) => {
  const pageNumers = []

  for (let i = 1; i <= Math.ceil(totalOrders / ordersPerPage); i += 1) {
    pageNumers.push(i)
  }

  return (
    <nav aria-label="Page navigation">
      <ul className="pagination justify-content-center mb-5">
        {pageNumers.map(num => {
          return (
            <li className="page-item" key={num}>
              <button
                type="button"
                onClick={() => paginate(num)}
                className="page-link"
              >
                {num}
              </button>
            </li>
          )
        })}
      </ul>
    </nav>
  )
}

export default Pagination
