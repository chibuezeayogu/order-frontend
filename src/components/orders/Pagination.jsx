import React from 'react'
import { Link } from 'react-router-dom'

const Pagination = ({ ordersPerPage, totalOrders, paginate }) => {
  const pageNumers = []

  for (let i = 1; i <= Math.ceil(totalOrders / ordersPerPage); i += 1) {
    pageNumers.push(i)
  }

  return (
    <nav aria-label="Page navigation">
      <ul className="pagination justify-content-center">
        {pageNumers.map(num => {
          return (
            <li className="page-item" key={num}>
              <Link
                onClick={() => paginate(num)}
                to={`/orders?page=${num}`}
                className="page-link"
              >
                {num}
              </Link>
            </li>
          )
        })}
      </ul>
    </nav>
  )
}

export default Pagination
