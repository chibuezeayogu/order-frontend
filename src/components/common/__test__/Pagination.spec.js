import React from 'react'
import { shallow } from 'enzyme'
import Pagination from '../Pagination'

const setUp = () => {
  const props = {
    ordersPerPage: 5,
    totalOrders: 20,
    paginate: jest.fn(),
    currentPage: 1
  }

  return shallow(<Pagination {...props} />)
}

describe('Pagination Component', () => {
  it('should render without crashing', () => {
    const wrapper = setUp()

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('nav')).toHaveLength(1)
    expect(wrapper.find('ul')).toHaveLength(1)
    expect(wrapper.find('li')).toHaveLength(4)
  })
})
