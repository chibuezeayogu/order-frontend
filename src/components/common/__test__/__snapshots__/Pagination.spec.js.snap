// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component should render without crashing 1`] = `
<nav
  aria-label="Page navigation"
>
  <ul
    className="pagination pagination-md justify-content-center mb-5"
  >
    <li
      className="page-item active"
      key="1"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
    </li>
    <li
      className="page-item "
      key="2"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
    </li>
    <li
      className="page-item "
      key="3"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
    </li>
    <li
      className="page-item "
      key="4"
    >
      <button
        className="page-link"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
    </li>
  </ul>
</nav>
`;
